addons/addon-base-raas/packages/base-raas-services/lib/study/study-permission-service.js

```diff
@@ -202,23 +208,18 @@ class StudyPermissionService extends Service {
           .delete(),
 
         // Remove study from user records
-        studyRecord.adminUsers.map(async principalIdentifier =>
-          this.upsertUserRecord(requestContext, {
-            studyId,
-            principalIdentifier,
-            addOrRemove: 'remove',
-            permissionLevel: 'admin',
-          }),
-        ),
-
-        studyRecord.readonlyUsers.map(async principalIdentifier =>
-          this.upsertUserRecord(requestContext, {
-            studyId,
-            principalIdentifier,
-            addOrRemove: 'remove',
-            permissionLevel: 'readonly',
-          }),
-        ),
+        permissionLevels.forEach(permissionLevel => {
+          if (studyRecord[`${permissionLevel}Users`]) {
+            studyRecord[`${permissionLevel}Users`].map(async principalIdentifier =>
```

Comment:
Since this mechanism is changing, I'd wait till https://github.com/awslabs/service-workbench-on-aws/pull/148/ is 
completed.

Reply:
Either of the pull requests would need to be rebased. Can you look at the PR and check if its good to go.